xmtp_mls/src/groups/mod.rs

```diff
@@ -132,6 +150,132 @@ where
         Ok(messages)
     }
 
+    fn get_message_id(
+        group_id: &[u8],
+        envelope_timestamp_ns: u64,
+        decrypted_message_bytes: &[u8],
+    ) -> Vec<u8> {
+        let mut id_vec = Vec::new();
+        id_vec.extend_from_slice(group_id);
+        id_vec.extend_from_slice(&envelope_timestamp_ns.to_be_bytes());
+        id_vec.extend_from_slice(decrypted_message_bytes);
+        sha256(&id_vec)
+    }
+
+    fn validate_message_sender(
+        &self,
+        openmls_group: &mut OpenMlsGroup,
+        decrypted_message: &ProcessedMessage,
+        envelope_timestamp_ns: u64,
+    ) -> Result<(String, Vec<u8>), MessageProcessingError> {
+        let mut sender_account_address = None;
+        let mut sender_installation_id = None;
+        if let Sender::Member(leaf_node_index) = decrypted_message.sender() {
+            if let Some(member) = openmls_group.member_at(*leaf_node_index) {
+                if member.credential == *decrypted_message.credential() {
```

Comment:
Credential should have `PartialEq` implemented on it, so you can just use `member.credential.eq(decrypte_message.credential())` without dereferencing.